### PR TITLE
Add general class for AnalysisResult and AnalysisStep

### LINKED ISSUE #34: Adding AnalysisResult
We want to add `AnalysisResult(Entity)` and methods for creating and linking these with `AnalysisStep`s inside `Analysis`.
Additionally these methods should be able to run from an ipynb.

Screenshot from 2024-09-18 11-43-39

### PR DESCRIPTION
**Pull request by ka-sarthak**

### CODE REVIEW

File: src/nomad_analysis/analysis_source.py
Comment (hampusnasstrom) [inline src/nomad_analysis/analysis_source.py:1]: Nice! This becomes quite a lot more compact now.

File: src/nomad_analysis/general/schema.py [hunk 1]
```diff
@@ -16,11 +16,36 @@
...
+)
 from nomad.metainfo import (
     SchemaPackage,
+    SubSection,
 )
 
 m_package = SchemaPackage()
 
 
+class AnalysisResult(Entity):
```
Comment (hampusnasstrom) [inline src/nomad_analysis/general/schema.py:31]: I guess we will rework this later to the Correlated Properties but this is fine for now.

**Review by hampusnasstrom — APPROVED** [review]
Great work Sarthak!

File: src/nomad_analysis/general/schema.py (hunk 1 above)
Comment (ka-sarthak) [reply src/nomad_analysis/general/schema.py:31]: Sounds good!

**Pull request merged**